src/sed/core/config.py

```diff
@@ -146,9 +159,19 @@ def parse_config(
 
     if not verify_config:
         return config_dict
-    # Run the config through the ConfigModel to ensure it is valid
-    config_model = ConfigModel(**config_dict)
-    return config_model.model_dump(exclude_unset=True, exclude_none=True)
+
+    try:
+        # Run the config through the ConfigModel to ensure it is valid
+        config_model = ConfigModel(**config_dict)
+        return config_model.model_dump(exclude_unset=True, exclude_none=True)
+    except ValidationError as e:
+        error_msg = (
+            "Invalid configuration file detected. The following validation errors were found:\n"
+        )
+        for error in e.errors():
+            error_msg += f"\n- {' -> '.join(str(loc) for loc in error['loc'])}: {error['msg']}"
+        logger.error(error_msg)
+        raise ConfigError(error_msg) from e
```

Comment:
Isn't it sufficient and actually better to just log the error here and raise the original error? Why creating a new error type?

Reply:
That's a good point. I just found that it makes thing very explicit but it can be removed.

Reply:
I have switched to a ValueError from ConfigError as suggested, as we are not trying to programatically catch the ConfigError so it's redundant.
It's possible to just log and not throw any error but then the testing has to be updated further.